docs/source/contributor-guide/development_environment.md

```diff
@@ -37,6 +37,22 @@ developing DataFusion in an isolated environment either locally or remote if des
 DataFusion is not a requirement by any means but is available for those where doing local development could be tricky
 such as with Windows and WSL2, those with older hardware, etc.
 
+To open the project in vscode you will need the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers). For codespaces you will need to [create a new codespace](https://github.com/apache/datafusion).
```

Comment:
thanks @dentiny for your contribution

Couple of notes
- referring to VSCode probably can be misleading as not all engineers using it. 
- Not sure why dev containers are needed for the development

Reply:
> referring to VSCode probably can be misleading as not all engineers using it.

Ok, I will remove. Add it because vscode is pretty popular.

> Not sure why dev containers are needed for the development

Devcontainer is a easy-setup datafusion provides, otherwise you need to manually prepare a clean environment.